### DISCUSSION TITLE
Pipeline Helper Function

### DISCUSSION THREAD

**Discussion by JordanDalton** [laravel/framework#46170] [Ideas]
Pipelines are quite powerful in Laravel. GIven how quick one may wish to utilize this feature I would like to propose the following helper:

```php
<?php


if( ! function_exists('pipe')) {

    function pipe($payload, $pipelines = [])
    {
        return app(\Illuminate\Pipeline\Pipeline::class)
                ->send($payload)
                ->through($pipelines)
                ->thenReturn();
    }
}
```

**Comment by newtonjob** [comment]
I think that the helper should return the `Pipeline` class itself. 

The `thenReturn` part should be chained externally by the developer.

```php
pipe(...)->thenReturn()
```

Especially because the user may want to return something other than the passable.

```php
pipe(...)->then(...)
```

**Reply by JordanDalton** [reply]
I considered this when I wrote the code. I'm not against it though.

**Comment by henzeb** [comment]
Just to be complete here:  there is already a package solving this and more. https://github.com/michael-rubel/laravel-enhanced-pipeline

**Comment by crynobone** [comment]
There's no plan to add more helpers at the moment you can already do the following:

```php
app('pipeline')->send($payload)...

// or 

use Illuminate\Support\Facades\Pipeline;

Pipeline::send($payload)...
```

**Comment by hafezdivandari** [comment]
Duplicate of #43429
